retrospective-update/functions.py

```
@@ -0,0 +1,351 @@
+
+        # Do the same for timeseries
+        chunks = xr.open_zarr(monthly_timeseries).chunks
+        (
+            months_ds
+            .chunk({"time": chunks["time"][0], "river_id": chunks["river_id"][0]})
+            .to_zarr(monthly_timeseries, mode='a', append_dim='time', consolidated=True)
+        )
+
+        # Update the hydrosos maps
+        date_range = pd.date_range(next_month, last_whole_month, freq='MS', inclusive='left')
+        update_hydrosos_maps(date_range, monthly_timesteps, hydro_sos_dir, credentials, CL)
```

Ruff (0.12.2)

Undefined name `update_hydrosos_maps`

(F821)